warp10/src/main/java/io/warp10/leveldb/LEVELDBCOMPACT.java

```diff
@@ -0,0 +1,102 @@
+//
+//   Copyright 2023  SenX S.A.S.
+//
+//   Licensed under the Apache License, Version 2.0 (the "License");
+//   you may not use this file except in compliance with the License.
+//   You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+//   Unless required by applicable law or agreed to in writing, software
+//   distributed under the License is distributed on an "AS IS" BASIS,
+//   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//   See the License for the specific language governing permissions and
+//   limitations under the License.
+//
+
+package io.warp10.leveldb;
+
+import java.util.concurrent.Callable;
+import java.util.concurrent.atomic.AtomicReference;
+
+import org.iq80.leveldb.Options;
+
+import io.warp10.script.NamedWarpScriptFunction;
+import io.warp10.script.WarpScriptException;
+import io.warp10.script.WarpScriptStack;
+import io.warp10.script.WarpScriptStackFunction;
+import io.warp10.standalone.Warp;
+import io.warp10.warp.sdk.Capabilities;
+
+public class LEVELDBCOMPACT extends NamedWarpScriptFunction implements WarpScriptStackFunction {
+
+  public LEVELDBCOMPACT(String name) {
+    super(name);
+  }
+
+  @Override
+  public Object apply(WarpScriptStack stack) throws WarpScriptException {
+
+    if (null == Capabilities.get(stack, WarpScriptStack.CAPABILITY_LEVELDB_ADMIN) && null == Capabilities.get(stack, WarpScriptStack.CAPABILITY_LEVELDB_COMPACT)) {
+      throw new WarpScriptException(getName() + " missing capability '" + WarpScriptStack.CAPABILITY_LEVELDB_ADMIN + "' or '" + WarpScriptStack.CAPABILITY_LEVELDB_COMPACT + "'.");
+    }
+
+    final AtomicReference<Throwable> error = new AtomicReference<Throwable>(null);
+
+    try {
+      WarpDB db = Warp.getDB();
+
+      if (null == db) {
+        throw new WarpScriptException(getName() + " can only be called when using LevelDB.");
+      }
```

Comment:
What about another standalone backend in a near future ?

Reply:
db is specific to LevelDB so will work.